src/lib.rs

```diff
@@ -109,3 +109,15 @@ pub mod traits {
 pub mod rational {
     pub use num_rational::*;
 }
+
+/// A value bounded by a minimum and a maximum
+///
+/// If input is less than min then min is returned, if input is greater than max then max is
+/// returned.  Otherwise input is returned.
+#[inline]
+pub fn clamp<T: PartialOrd + Copy>(input: T, min: T, max: T) -> T {
+    debug_assert!(min < max, "min must be less than max");
+    if input <= min {min}
+    else if input >= max {max}
+    else {input}
```

Comment:
This formatting is non-standard -- please run it through rustfmt.

I think it would be slightly preferable to return `input` when it is equal to the boundaries.  This only matters for cases similar to stable sorting, where there may be internal data that's not part of the ordering.  So just compare `input < min` and `input > max`.